smart-contracts/contracts-common/concordium-contracts-common-derive/src/attribute.rs

```diff
@@ -659,9 +662,11 @@ pub fn receive_worker(attr: TokenStream, item: TokenStream) -> syn::Result<Token
                 if let Ok(state) = DeserialWithState::deserial_with_state(&state_api, &mut state_api.lookup_entry(&[]).unwrap_abort()) {
                     let mut state_builder = StateBuilder::open(state_api);
                     let mut host = ExternHost { state, state_builder };
-                    match #fn_name(&ctx, #host_ref, #(#fn_optional_args, )*) {
+                    let result = #fn_name(&ctx, #host_ref, #(#fn_optional_args, )*);
```

Comment:
**Suggestion (non-blocking):** Add a comment or some test case, ensuring we don't inline this again in the future.

Reply:
The problem is that the test for this would have to be in concordium-rust-smart-contracts repository which is not ideal.

I'll add a comment though.

Reply:
Added.